lib/server.js

```diff
@@ -675,12 +675,18 @@ class WebServer {
             //  alarm_arm_away, alarm_arm_home, alarm_arm_night, alarm_arm_custom_bypass, alarm_disarm (code will be sent)
             this.log.debug(data.service + ': ' + id + ' = ' + data.service_data.code ? 'XXX' : 'none');
             this.adapter.getForeignObject(id, (err, obj) => {
-                if (obj.native.alarm_code && obj.native.alarm_code.toString() !== data.service_data.code.toString()) {
+                this.log.debug("native" + JSON.stringify(obj.native));
+                if (data.service_data == null || data.service_data.code == null) {
+                    this._sendResponse(ws, data.id, {result: false, error: 'code is empty'});
+                } else if (!obj.native.alarm_code || !data.service_data.code) {
```

Comment:
The warning is missleading if `!data.service_data.code` resolves to `true`. Why is this check here, anyway? The null check from above and the !== check from below should be sufficient, or not?